test/shutdown_test.py

```diff
@@ -64,6 +64,7 @@ async def _mocked_logs_loop(client: Client, app_id: str):
         await t
 
     with Client(servicer.client_addr, api_pb2.CLIENT_TYPE_CLIENT, credentials) as client:
+        await client.hello.aio()  # TODO(erikbern): why is this needed?
```

Comment:
I looked at this the other week I think - if I'm not mistaken this was due to grpclib's channels being lazily connected by the first rpc call. Removing the hello from the client opening sequence made it instead set up the channel in the user space event loop used by `_mocked_logs_loop` here, which invalidates this test (but probably has no real effect on users of the SDK).

This test was mostly just in place for internal usage warnings during development anyway (the warnings etc.).
An alternative fix would be to make sure that `Client._open` still calls `channel.__connect__` to set up the connection (or something like that).

Not related to this PR, but I think we should just prohibit cross-event loop usage of the client altogether instead of trying to be accomodating here.

Reply:
> An alternative fix would be to make sure that `Client._open` still calls `channel.__connect__` to set up the connection (or something like that).

This seems useful in itself, I can take a look at this!

Reply:
Thanks for explanation!